### PR TITLE
 issue: 1582189 Add Device Memory (Memic) support for upstream

### PR DESCRIPTION
**Pull request by liranoz12**

### CODE REVIEW

File: src/vma/dev/dm_mgr.cpp [hunk 1]
```diff
@@ -217,17 +214,15 @@ bool dm_mgr::copy_data(struct mlx5_wqe_data_seg* seg, uint8_t* src, uint32_t len
...
-	};
+	// Initialize memcopy attributes
+	memset(&memcpy_attr, 0, sizeof(memcpy_attr));
+	vma_ibv_init_memcpy_dm(memcpy_attr, src, m_head, length_aligned_8);
+
+	// Copy data into the On Device Memory buffer.
+	if (vma_ibv_memcpy_dm(m_p_ibv_dm, &memcpy_attr)) {
+		dm_logfunc("Failed to memcopy data into the memic buffer %m");
+		return false;
+	}
```
Comment (igor-ivanov) [inline src/vma/dev/dm_mgr.cpp]: It is not clear why verbs api is not used in original implementation. But if changes in the pull request more readable.

**Comment by liranoz12** [comment]
bot:retest

**Comment by liranoz12** [comment]
bot:retest

**Comment by liranoz12** [comment]
bot:retest

**Comment by liranoz12** [comment]
bot:retest

**Pull request merged**